internal/driven/storage/redis/queststrg/storage.go

```diff
@@ -0,0 +1,113 @@
+package queststrg
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"math/rand"
+	"time"
+
+	"github.com/ghazlabs/hex-mathrush/internal/core"
+	"github.com/go-redis/redis/v8"
+	"gopkg.in/validator.v2"
+)
+
+type Storage struct {
+	redisClient *redis.Client
+}
+
+type Config struct {
+	RedisClient *redis.Client `validate:"nonnil"`
+}
+
+var questList []core.Question
+
+func New(cfg Config) (*Storage, error) {
+	err := validator.Validate(cfg)
+	if err != nil {
+		return nil, err
+	}
+	s := &Storage{redisClient: cfg.RedisClient}
+	//insert qurestion berupa list of Question struct
+	questList = []core.Question{
+		{
+			Problem:      "1 + 2",
+			Choices:      []string{"1", "2", "3"},
+			CorrectIndex: 3,
+		},
+		{
+			Problem:      "1 + 1",
+			Choices:      []string{"1", "2", "3"},
+			CorrectIndex: 2,
+		},
+		{
+			Problem:      "2 + 1",
+			Choices:      []string{"1", "2", "3"},
+			CorrectIndex: 3,
+		},
+		{
+			Problem:      "3 - 2",
+			Choices:      []string{"1", "2", "3"},
+			CorrectIndex: 1,
+		},
+		{
+			Problem:      "2 - 1",
+			Choices:      []string{"1", "2", "3"},
+			CorrectIndex: 1,
+		},
+		{
+			Problem:      "2 + 2 - 1",
+			Choices:      []string{"1", "2", "3"},
+			CorrectIndex: 3,
+		},
+		{
+			Problem:      "1 + 1 + 1",
+			Choices:      []string{"1", "2", "3"},
+			CorrectIndex: 3,
+		},
+		{
+			Problem:      "3 - 1",
+			Choices:      []string{"1", "2", "3"},
+			CorrectIndex: 2,
+		},
+		{
+			Problem:      "2 + 1 - 2",
+			Choices:      []string{"1", "2", "3"},
+			CorrectIndex: 1,
+		},
+		{
+			Problem:      "1 + 1 - 1",
+			Choices:      []string{"1", "2", "3"},
+			CorrectIndex: 1,
+		},
+	}
+	str, err := json.Marshal(questList)
+	if err != nil {
+		log.Fatalf("unable to marshaling intial question: %v", err)
+	}
+	err = s.redisClient.Set(context.Background(), "questions", string(str), 0).Err()
+	if err != nil {
+		log.Fatalf("unable to set initial question: %v", err)
+	}
+	return s, nil
+}
+
+func (s *Storage) GetRandomQuestion(ctx context.Context) (*core.Question, error) {
+
+	//res, err := rdb.Do(ctx, "set", "key", "value").Result()
+	//val2, err := rdb.Get(ctx, "key2").Result()
+	var qList []core.Question
+	val, err := s.redisClient.Get(ctx, "questions").Result()
+	if err != nil {
+		log.Fatalf("unable to get question: %v", err)
```

Comment:
Never use `log.Fatalf()` inside component function.

The reason is because `log.Fatalf()` will make the whole program exit immediately. So it should only be called on `main.go`.

In here what you should do is returning the error gracefully like this:

```golang
return fmt.Errorf("unable to get questions due: %w", err)
```